File: app/src/main/java/app/akilesh/nex/fragments/DeviceFragment.java
package app.akilesh.nex.fragments;

import android.os.Build;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import app.akilesh.nex.R;

import static android.content.ContentValues.TAG;


public class DeviceFragment extends Fragment {

    public static DeviceFragment newInstance() {
        return new DeviceFragment();
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }


    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.device_fragment, container, false);

        String brand = Build.BRAND;
        TextView deviceBrand = view.findViewById(R.id.brand);
        deviceBrand.setText(String.format("%s", brand));

        String model = Build.MODEL;
        TextView deviceModel = view.findViewById(R.id.model);
        deviceModel.setText(String.format("%s", model));

        String code = Build.DEVICE;
        TextView deviceCodeName = view.findViewById(R.id.codeName);
        deviceCodeName.setText(String.format("%s", code));

        String buildVer = "Unknown";

        try {
            Process process = Runtime.getRuntime().exec("su");
            InputStream in = process.getInputStream();
            OutputStream out = process.getOutputStream();
            String cmd = "[ -r /data/misc/box/report/PREV_VER ] && cat /data/misc/box/report/PREV_VER";
            out.write(cmd.getBytes());
            out.flush();
            out.close();
            process.waitFor();

            if (process.exitValue() != 0) {
                Log.e(TAG,"Failed to obtain root");
            }
            else {
                int ch;
                StringBuilder sb = new StringBuilder();
                while((ch = in.read()) != -1)
                    sb.append((char)ch);
                buildVer = sb.toString();
            }

            buildVer = buildVer.trim();

        } catch (IOException e) {
                Log.e(TAG, "IOException, " + e.getMessage());

        } catch (InterruptedException e) {
                Log.e(TAG, "InterruptedException, " + e.getMessage());
        }

        TextView build = view.findViewById(R.id.buildVersion);
        if(!buildVer.isEmpty()){
        build.setText(String.format("%s", buildVer));
        }
        else  build.setText(String.format("%s", "Not found"));
        return view;
    }


    @Override
    public void onDetach() {
        super.onDetach();
    }


    public interface OnFragmentInteractionListener {
    }

}

File: app/src/main/java/app/akilesh/nex/fragments/DeviceFragment.kt
package app.akilesh.nex.fragments

import android.os.Build
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import androidx.fragment.app.Fragment
import app.akilesh.nex.R
import com.topjohnwu.superuser.Shell
import kotlinx.android.synthetic.main.fragment_device.*
import java.io.File


class DeviceFragment : Fragment(){
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_device, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        brand.text = Build.BRAND
        model.text = Build.MODEL
        codeName.text = Build.DEVICE
        buildFingerprint.text = Build.FINGERPRINT
        val firmwareVersionPath = "/proc/fver"
        val skuidPath = "/proc/cda/skuid"
        var files  = false
        if(File(firmwareVersionPath).exists() && File(skuidPath).exists())
           files = true

        val outputs = mutableListOf<String>()
        Shell.sh("getprop ro.build.ab_update").to(outputs).exec()
        if(outputs.component1() == "true") {
            val out = Shell.sh("getprop ro.boot.slot_suffix").exec().out
            slot.text = out.toString().drop(2).dropLast(1)
        }

        if(Shell.rootAccess() && files) {
            Shell.su("[ -f $firmwareVersionPath ] && head -n 1 $firmwareVersionPath").to(outputs).exec()
            buildVersion.text = outputs.component2().substring(4, 23)

            Shell.su("[ -f $skuidPath ] && head -n 1 $skuidPath").to(outputs).exec()
            skuid.text = outputs.component3()
        }
        else {
            Shell.sh("getprop ro.build.version.incremental").to(outputs).exec()
            buildVersion.text = outputs.component2()

            Shell.sh("getprop ro.cda.skuid.id").to(outputs).exec()
            skuid.text = outputs.component3()
        }

        Shell.sh("getprop ro.build.version.security_patch").to(outputs).exec()
        asp.text = outputs.component4()

        Shell.sh("getprop ro.vendor.build.security_patch").to(outputs).exec()
        vsp.text = outputs.component5()



    }
    companion object {
        internal const val TAG = "DeviceFragmentTag"
    }

}
